File: src/providers/cli/codex.js
/**
 * @module providers/cli/codex
 * @description OpenAI Codex CLI adapter.
 *
 * Invokes agents via: `codex --prompt <input> --full-auto`
 * Agent name is prepended to the input as context.
 */

const { CliAdapterBase } = require('../base/cli-adapter-base');

class CodexAdapter extends CliAdapterBase {
  get name() { return 'codex'; }
  get label() { return 'Codex CLI'; }
  get defaultCommand() { return 'codex'; }

  buildArgs(agentName, input, config) {
    // Codex doesn't have an --agent flag, so we prepend the agent role to the prompt
    const prompt = `[Agent: ${agentName}]\n${input}`;
    const args = ['--prompt', prompt, '--full-auto'];

    if (config.model) {
      args.push('--model', config.model);
    }
    if (config.extraArgs && Array.isArray(config.extraArgs)) {
      args.push(...config.extraArgs);
    }

    return args;
  }

  parseOutput(stdout, stderr, exitCode) {
    return {
      success: exitCode === 0,
      output: stdout,
      error: exitCode !== 0 ? (stderr || `Exited with code ${exitCode}`) : null,
    };
  }
}

module.exports = new CodexAdapter();

File: src/providers/cli/codex.py
"""Codex CLI adapter."""

from src.providers.base import CliAdapterBase


class CodexAdapter(CliAdapterBase):
    name = "codex"
    label = "Codex CLI"
    default_command = "codex"
    agent_dir = ".codex/agents"
    config_dir = ".codex"
    rules_file_name = "AGENTS.md"

    def build_args(self, agent_name, input_text, config):
        prompt = f"[Agent: {agent_name}]\n{input_text}"
        args = ["--prompt", prompt, "--full-auto"]
        if config.get("model"):
            args.extend(["--model", config["model"]])
        args.extend(config.get("extra_args") or [])
        return args

    def parse_output(self, stdout, stderr, exit_code):
        return {
            "success": exit_code == 0,
            "output": stdout,
            "error": stderr or f"Exited with code {exit_code}" if exit_code != 0 else None,
        }


adapter = CodexAdapter()
